src/nex_types.rs

```diff
@@ -0,0 +1,446 @@
+                "sid" => station_url.sid = value,
+                "CID" => station_url.cid = value,
+                "PID" => station_url.pid = value,
+                "type" => station_url.transport_type = value,
+                "RVCID" => station_url.rvcid = value,
+                "natm" => station_url.natm = value,
+                "natf" => station_url.natf = value,
+                "upnp" => station_url.upnp = value,
+                "pmp" => station_url.pmp = value,
+                "probeinit" => station_url.probe_init = value,
+                "PRID" => station_url.prid = value,
+                _ => {}
+            }
+        }
+        station_url
+    }
+
+    pub fn encode_to_string(&self) -> String {
+        let mut fields = Vec::new();
+
+        if let Some(address) = &self.address {
+            fields.push(address.to_string());
+        }
+
+        if let Some(port) = &self.port {
+            fields.push(port.to_string());
+        }
+
+        if let Some(stream) = &self.stream {
+            fields.push(stream.to_string());
+        }
+
+        if let Some(sid) = &self.sid {
+            fields.push(sid.to_string());
+        }
+
+        if let Some(cid) = &self.cid {
+            fields.push(cid.to_string());
+        }
+
+        if let Some(pid) = &self.pid {
+            fields.push(pid.to_string());
+        }
+
+        if let Some(transport_type) = &self.transport_type {
+            fields.push(transport_type.to_string());
+        }
+
+        if let Some(rvcid) = &self.rvcid {
+            fields.push(rvcid.to_string());
+        }
+
+        if let Some(natm) = &self.natm {
+            fields.push(natm.to_string());
+        }
+
+        if let Some(natf) = &self.natf {
+            fields.push(natf.to_string());
+        }
+
+        if let Some(upnp) = &self.upnp {
+            fields.push(upnp.to_string());
+        }
+
+        if let Some(pmp) = &self.pmp {
+            fields.push(pmp.to_string());
+        }
+
+        if let Some(probe_init) = &self.probe_init {
+            fields.push(probe_init.to_string());
+        }
+
+        if let Some(prid) = &self.prid {
+            fields.push(prid.to_string());
+        }
+
+        let fields_string = fields.join(";");
+        let scheme = if let Some(scheme) = &self.scheme {
+            scheme.to_string()
+        } else {
+            "".to_string()
+        };
+
+        format!("{}:/{}", scheme, fields_string)
+    }
+}
+
+struct Result {
+    result_code: u32
+}
+
+impl Result {
+    pub fn new_from_code(result_code: u32) -> Self {
+        Self {
+            result_code
+        }
+    }
+
+    pub fn extract_from_stream<T: Reader>(&mut self, stream: &mut StreamIn<T>) -> core::result::Result<(), &'static str> {
+        self.result_code = stream
+            .read_stream_le::<u32>()
+            .map_err(|_| "Result code could not be read")?
+            .try_into()
+            .map_err(|_|"Result code size does not fit into u32")?;
```

Comment:
Since `StreamIn` can read u32s, we don't need `try_into`.  The `try_into` is confusing rust and causing the need to specify the types, so removing `try_into` means we can remove the types as well:
```suggestion
        self.result_code = stream
            .read_stream_le()
            .map_err(|_| "Result code could not be read")?;
```